src/components/InvertedFlatList/BaseInvertedFlatList.js

```diff
@@ -118,36 +112,34 @@ class BaseInvertedFlatList extends Component {
      *
      * @return {React.Component}
      */
-    renderItem({item, index}) {
-        if (this.props.shouldMeasureItems) {
-            return <View onLayout={({nativeEvent}) => this.measureItemLayout(nativeEvent, index)}>{this.props.renderItem({item, index})}</View>;
-        }
-
-        return this.props.renderItem({item, index});
-    }
-
-    render() {
-        return (
-            <FlatList
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...this.props}
-                ref={this.props.innerRef}
-                renderItem={this.renderItem}
-                // Native platforms do not need to measure items and work fine without this.
-                // Web requires that items be measured or else crazy things happen when scrolling.
-                getItemLayout={this.props.shouldMeasureItems ? this.getItemLayout : undefined}
-                windowSize={15}
-
-                // Commenting the line below as it breaks the unread indicator test
-                // we will look at fixing/reusing this after RN v0.72
-                // maintainVisibleContentPosition={{minIndexForVisible: 0, autoscrollToTopThreshold: 0}}
-            />
-        );
-    }
+    const renderItemFromProp = useCallback(
+        ({item, index}) => {
+            if (shouldMeasureItems) {
+                return <View onLayout={({nativeEvent}) => measureItemLayout(nativeEvent, index)}>{renderItem({item, index})}</View>;
+            }
+
+            return renderItem({item, index});
+        },
+        [shouldMeasureItems, measureItemLayout, renderItem],
+    );
+
+    return (
+        <FlatList
+            // eslint-disable-next-line react/jsx-props-no-spreading
```

Comment:
Is there a reason we're suppressing this here instead of updating the code to comply with this rule?

Reply:
I work on App much much less than you all, but I actually don't think ignoring the rule is better, and I'd love for us to consider passing all props explicitly
- It makes the code easier to understand, because you know what's being passed instead of going up the component hierarchy
- It doesn't allow us to send extra props that aren't needed (maybe there are ways this is prevented that I don't know about)
- Remove the rule would be better than having it and ignoring it

Reply:
> When not to use the rule: If you are not using JSX or have lots of props to be passed or the props spreading is used inside HOC.

I think this is close, except the "have lots of props to be passed" is kind of misleading. The case where it's OK to use the spread operator is when you have a wrapper component that can pass _any vast number_ of props and there is no way to know what those props will be.

This is why it's OK to be used inside an HOC (because it's generic and any component can be passing props).

I think there is another case worth mentioning and that is when there is a wrapper around a native-built or a third-party component. I don't think we would want to try and explicitly pass every prop that is available to those components (ie. we might not know all the props, or the props might change with library updates that we would not catch and know about).

That's why I think there is a pretty good case for using the spread operator here.

For the majority of cases, we shouldn't use the spread operator and therefore it's still valuable for the ESlint rule to be there. It gives us a chance to do a thorough review of all the exceptions.

Reply:
@cead22 how do we proceed with this? should we not suppress the warning? Also AFFAIK, there are many other exceptional cases as this for HOCs.

Reply:
Yeah that makes sense. We can continue that discussion elsewhere without blocking this PR on it